packages/desktop-client/src/components/util/GenericInput.jsx

```diff
@@ -209,15 +212,30 @@ export function GenericInput({
           />
         );
       } else {
-        content = (
-          <Input
-            inputRef={inputRef}
-            defaultValue={value || ''}
-            placeholder="nothing"
-            onEnter={e => onChange(e.target.value)}
-            onBlur={e => onChange(e.target.value)}
-          />
-        );
+        switch (type) {
+          case 'number':
+            content = (
+              <AmountInput
+                inputRef={inputRef}
+                value={amountToInteger(value)}
+                placeholder="0.00"
+                onEnter={e => onChange(e.target.value)}
+                onBlur={e => onChange(e.target.value)}
+                onUpdate={newValue => onChange(integerToAmount(newValue))}
+              />
+            );
+            break;
+          default:
+            content = (
+              <Input
+                inputRef={inputRef}
+                defaultValue={value || ''}
+                placeholder="nothing"
+                onEnter={e => onChange(e.target.value)}
+                onBlur={e => onChange(e.target.value)}
```

Comment:
You can just use onUpdate here which fires on blur

Reply:
ahaha thanks! I had just noticed this as well, fixed